skipruntime-ts/ts/src/skipruntime_api.ts

```diff
@@ -366,6 +366,11 @@ export interface EagerCollection<K extends TJSON, V extends TJSON> {
    */
   sliced(ranges: [K, K][]): EagerCollection<K, V>;
 
+  /**
+   * Create a new eager collection by keeping the given number of the first elements.
+   */
+  take(limit: int): EagerCollection<K, V>;
```

Comment:
Is `C.take(N)` supposed to take the first `N` values in `C`, or all the values of the first `N` keys in `C`? I ask since, from the code, it seems like the latter, but the docs read as the former to me.

Reply:
Currently it's the latter indeed. Real usage will tell whether we need more flexibility

Reply:
Looking over this PR again, I don't know if it is clear what "first" means. As far as I understand `C.take(N)` will produce a collection with `N` elements if `C` already has `N` or more elements, and the elements included will be those with the smallest keys according to the order on the underlying Key type. But if one of those elements is then removed from `C`, will it be repaced in the result of take by a new one? Or will the result of take have fewer than `N` elements until more are added to `C`? I haven't dug through the implementation of filter in skstore deeply enough to answer these questions. Any insights @pikatchu ?